plugins/modules/redhat_subscription.py

```diff
@@ -865,7 +875,7 @@ def update_subscriptions_by_pool_ids(self, pool_ids):
             existing_pools[pool_id] = quantity_used
 
             quantity = pool_ids.get(pool_id, 0)
-            if quantity is not None and quantity != quantity_used:
+            if quantity is not None and quantity != quantity_used and not keep_consumed_pools:
                 serials_to_remove.append(p.Serial)
 
         serials = self.unsubscribe(serials=serials_to_remove)
```

Comment:
i'd guard this `keep_consumed_pools` as well: right now, it seems to work because `unsubscribe()` does nothing when passed with an empty; since that behaviour contradicts the docstring of that function itself, i'd not rely on that